pkg/gen/input/makefile/internal/file/makefile.go

```diff
@@ -0,0 +1,84 @@
+package file
+
+import (
+	"github.com/giantswarm/devctl/pkg/gen/input"
+)
+
+func NewMakefileInput() input.Input {
+	i := input.Input{
+		Path:         "Makefile",
+		TemplateBody: makefileTemplate,
+		TemplateData: map[string]interface{}{},
+	}
+
+	return i
+}
+
+var makefileTemplate = `# DO NOT EDIT. Generated with:
+#
+#    devctl gen makefile
+#
+
+APPLICATION    := $(shell basename $(shell go list .))
+BUILDTIMESTAMP := $(shell date -u '+%FT%TZ')
+GITSHA1        := $(shell git rev-parse --verify HEAD)
+OS             := $(shell go env GOOS)
+SOURCES        := $(shell find . -name '*.go')
+VERSION        := $(shell architect project version)
+LDFLAGS        ?= -w -linkmode 'auto' -extldflags '-static' \
+  -X '$(shell go list .)/pkg/project.buildTimestamp=${BUILDTIMESTAMP}' \
+  -X '$(shell go list .)/pkg/project.gitSHA=${GITSHA1}' \
+  -X '$(shell go list .)/pkg/project.version=${VERSION}'
+.DEFAULT_GOAL := build
+
+.PHONY: build build-darwin build-linux
+## build: builds a local binary
+build: $(APPLICATION)
+## build-darwin: builds a local binary for darwin/amd64
+build-darwin: $(APPLICATION)-darwin
+## build-linux: builds a local binary for linux/amd64
+build-linux: $(APPLICATION)-linux
+
+$(APPLICATION): $(APPLICATION)-$(OS)
+	cp -a $< $@
+
+$(APPLICATION)-%: $(SOURCES)
+	GOOS=$* GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o $@ .
+
+.PHONY: install
+## install: install the application
+install:
+	go install -ldflags "$(LDFLAGS)" .
+
+.PHONY: run
+## run: runs go run main.go
+run:
+	go run -ldflags "$(LDFLAGS)" -race .
+
+.PHONY: clean
+## clean: cleans the binary
+clean:
+	rm -f $(APPLICATION)*
+	go clean
+
+.PHONY: lint
+## lint: runs golangci-lint
+lint:
+	golangci-lint run -E gosec -E goconst --timeout=15m ./...
+
+.PHONY: test
+## test: runs go test with default values
+test:
+	go test -ldflags "$(LDFLAGS)" -race -cover ./...
```

Comment:
Do we need `-cover` flag there? We don't use it in the CI and I'm not sure it's useful as we are not super fussed about test coverage in general. We also don't have `-race` but I think that's ok if it doesn't take ages to run it in the CI machine.

https://github.com/giantswarm/architect-orb/blob/d1535b34fad4df844df9f183325fde5c24b28ba8/src/commands/go-test.yaml#L23

Reply:
@kopiczko I dropped it but left -race. I think this shouldn't increase the amount of time when running it.

Reply:
Fine with me.